src/azul/service/responseobjects/storage_service.py

```diff
@@ -126,6 +137,19 @@ def __complete(self):
 
         try:
             self.mp_upload.complete(MultipartUpload={"Parts": [part.to_dict() for part in self.parts]})
+            if self.filename:
```

Comment:
You might want to consider adding this to an `else` branch after the `except`. But I am not sure since this could also raise an error.

Reply:
Yes, the code you added should be in a separate `try:`, nested in the `else:` branch of this `try:`.

And you must now handle eventual consistency. If put_object_tagging claims that the object does not exist you should wait 5s and retry for up to a 1 min.